tests/test_base_device.py

```diff
@@ -1204,9 +1208,53 @@ def test_generate_samples(self, num_wires, num_shots):
         # nothing else is in samples
         assert [s for s in samples if not s in np.array([exp_res0, exp_res1])] == []
 
+    @pytest.mark.parametrize("observable", [qml.X(0), qml.Y(1), qml.Z(2)])
+    @pytest.mark.parametrize(
+        "measurement",
+        [
+            lambda obs: qml.expval(obs),
+            lambda obs: qml.counts(obs),
+            lambda obs: qml.var(obs),
+        ],  # pylint: disable=unnecessary-lambda
+    )
+    @pytest.mark.parametrize("angle", [np.pi / 2, np.pi / 3, np.pi / 4])
+    def test_behavior_for_use_primitives(self, angle, measurement, observable):
+        """Tests that the format for results from ``use_primitives=True`` and ``use_primitives=False``
+        are correct. ``qml.probs()`` is excluded due to the fact that it does not take in an
```

Comment:
qml.probs() will be addressed in a [separate PR](https://github.com/PennyLaneAI/pennylane-qiskit/pull/529)

Reply:
We could use the keyword argument like `lambda obs: qml.measurement(op=obs)`, which should work for `probs` as well, right?

Reply:
Yes that works. I didn't write a test because right now the output from probs() is already different from the output from pennylane -- this has its own PR so I'll put it off for that PR :)